### DISCUSSION TITLE
No premissions to make directories after upgrade to 1.3.1

### DISCUSSION THREAD

**Discussion by Blaiz0** [Zibbp/ganymede#218] [Q&A] [answered]
Hi, 
Im running ganymede on a dockcontainer on my synology nas, its been working flawlessly for about half a year.
A few days ago i decided to update the software by SSHing into the NAS and using "sudo docker-compose pull" and then "sudo docker-compose up".

Now im getting a fault message in the queue, on create folder
and this error in ganymede-api:
```
{"level":"error","error":"mkdir /vods/jinnytty/2023-06-03-42265219835-live-b3804017-01aa-11ee-9c66-0242ac150005: permission denied","time":"2023-06-03T03:05:16+02:00","message":"error creating vod folder"} | stderr
```

The thing is that a added a few new streamers, and tried to archive them manually, it works.
The new streamers work when archiving automatically too. Tried to manually archive a streamer i have archived previously, and the same error message comes immediately. 
It seems like it does not have write access to the folders that were created before i updated. 

Does anyone have any idea how i can fix this? I doubt it is related to the changes in the software, more how i updated the containers.

I did check in the NAS GUI that my username and the admin user have read and write access to the VOD folder. But im not 100% sure what user the container is running from, but i think it should be my user or the admin account since i used sudo command when creating the container.

**Comment by Zibbp** [comment, marked as answer]
A few months ago I made the container rootless. You will probably need to specify a PUID and PGID environment variable. Set these to the user that has access to the vods folder. More information can be found here https://github.com/Zibbp/ganymede#rootless. When initially starting the API container it will print out what user it's being run as. The default has a uid/gid of `911`. Let me know if that still doesn't work.

**Comment by Blaiz0** [comment]
I see, it is uid 911 now! And as far as i know there is no user with that id, or at least they won't have access to the folder.
Did an "id" to find my own uid/gid and added that as environment variables.

I was forced to use the gid 101 administrators, even though 100 users should have write access. Guessing this is probably some synology thing.

My bad for not reading the documentation carefully enough. Only looked at the yml file to see if there was any changes.
Thanks for your awesome project and incredible support!  I really appreciate this 👍
